src/components/views/settings/tabs/user/HelpUserSettingsTab.js

```diff
@@ -80,6 +80,14 @@ export default class HelpUserSettingsTab extends React.Component {
         });
     };
 
+    _onMarkAllRoomsAsRead = (e) => {
+        MatrixClientPeg.get().getRooms().forEach(function(element) {
+            MatrixClientPeg.get().sendReadReceipt(function(evs) {
+                return evs[evs.length - 1]
+            }(element.getLiveTimeline()._events), {});
```

Comment:
shouldn't be accessing private variables
https://github.com/matrix-org/matrix-react-sdk/pull/2959/files#diff-beb48c1c7c8677b581a5b4177c817f28R661 shows an alternative

Reply:
Didn't realize #2959 existed. Let's merge that and just add another button to there that marks all rooms as read.